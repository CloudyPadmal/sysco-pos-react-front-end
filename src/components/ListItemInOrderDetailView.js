import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

class ListItemInOrderDetailView extends Component {

    render() {
        let { productTitle, description, quantity, price } = this.props.singleItem;
        return (
            <li className="list-group-item d-flex justify-content-right" style={{ margin: '0px', padding: '0px', width: '100%' }}>
                <div className="row" style={{ margin: '0px', width: '100%' }}>
                    <div className="col-9 card">
                        <div className="card-body" style={{ padding: '5px 0px' }}>
                            <h6 className="card-title d-flex justify-content-right">{productTitle}</h6>
                            <p className="card-subtitle text-muted d-flex justify-content-right">{description}</p>
                        </div>
                    </div><div className="col-1 card" style={{ padding: '0px', textAlign: 'center' }}>
                        <div className="card-body d-flex align-items-center">
                            <b>{quantity}</b>&nbsp;Nos
                        </div>
                    </div>
                    <div className="col-2 card text-right">
                        <p className="card-text" style={{ margin: '5px 0 0 0' }}>@ {price.toFixed(2)}</p>
                        <p className="card-link" style={{ margin: '0px' }}><b>Rs. {(quantity * price).toFixed(2)}</b></p>
                    </div>
                </div>
            </li>
        );
    }
}

ListItemInOrderDetailView.propTypes = {
    ISLOGGEDIN: PropTypes.bool.isRequired,
    CURRENTORDER: PropTypes.object.isRequired,
    PASSKEY: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
    PASSKEY: state.uac.PASSKEY,
    ISLOGGEDIN: state.uac.ISLOGGEDIN,
    CURRENTORDER: state.cor.CURRENTORDER
});

export default connect(mapStateToProps, null)(ListItemInOrderDetailView);