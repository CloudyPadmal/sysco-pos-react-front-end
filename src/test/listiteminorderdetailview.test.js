import React from 'react';
import { mountWrap } from './enzyme';

import ListItemInOrderDetailView from '../components/ListItemInOrderDetailView';

describe('List Item In Order', () => {

    it('displays a list of items to edit order', () => {
        const wrapper = mountWrap(<ListItemInOrderDetailView
            key={123123}
            singleItem={{ price: 34.33 }}
        />);
        expect(wrapper).toMatchSnapshot();
    });
});