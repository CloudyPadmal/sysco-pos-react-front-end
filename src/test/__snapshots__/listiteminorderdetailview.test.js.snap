// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item In Order displays a list of items to edit order 1`] = `
<ListItemInOrderDetailView
  singleItem={
    Object {
      "price": 34.33,
    }
  }
>
  <li
    className="list-group-item d-flex justify-content-right"
    style={
      Object {
        "margin": "0px",
        "padding": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className="row"
      style={
        Object {
          "margin": "0px",
          "width": "100%",
        }
      }
    >
      <div
        className="col-9 card"
      >
        <div
          className="card-body"
          style={
            Object {
              "padding": "5px 0px",
            }
          }
        >
          <h6
            className="card-title d-flex justify-content-right"
          />
          <p
            className="card-subtitle text-muted d-flex justify-content-right"
          />
        </div>
      </div>
      <div
        className="col-1 card"
        style={
          Object {
            "padding": "0px",
            "textAlign": "center",
          }
        }
      >
        <div
          className="card-body d-flex align-items-center"
        >
          <b />
          Â Nos
        </div>
      </div>
      <div
        className="col-2 card text-right"
      >
        <p
          className="card-text"
          style={
            Object {
              "margin": "5px 0 0 0",
            }
          }
        >
          @ 
          34.33
        </p>
        <p
          className="card-link"
          style={
            Object {
              "margin": "0px",
            }
          }
        >
          <b>
            Rs. 
            NaN
          </b>
        </p>
      </div>
    </div>
  </li>
</ListItemInOrderDetailView>
`;
